client/my-sites/themes/validate-filters.js

```diff
@@ -81,3 +83,33 @@ export function sortFilterTerms( context, terms ) {
 		.sort()
 		.map( ( filter ) => getThemeFilterTermFromString( context.store.getState(), filter ) );
 }
+
+export function fetchAndValidateVerticalsAndFiltersIfLoggedIn( context, next ) {
+	const state = context.store.getState();
+	const currentUser = getCurrentUser( state );
+
+	if ( ! currentUser ) {
+		return next();
+	}
```

Comment:
You don't need these logged-in/out checks if the `selectSiteIfLoggedIn` and `redirectLoggedOut` handlers terminate early and don't call `next`. `redirectLoggedOut` already behaves that way. `selectSiteIfLoggedIn` needs to be fixed.

Then both `fetchAndValidate` handlers can be merged into one.